Added Support for Tv's with Older Firmware

zerbzman opened the pull request:
Added support for Tv's with firmware pre 4.0. This requires a different port number when communicating with the tv. 

`On firmware versions older than 4.0 the API server runs on port 9000, on 4.0 and newer the API server runs on port 7345. Both are using https and will not respond to http.` - [Vizio SmartCast API (2016+ Models)](https://github.com/exiva/Vizio_SmartCast_API)

zerbzman committed:
Added Support with Tv's with Older Firmware

Added support for Tv's with firmware pre 4.0.

josephdadams commented:
@zerbzman Thanks for contributing!

I think this will break existing users when they upgrade. The config setting “firmware” will be undefined (which is not === 1) which means it will default to the pre 4.0 settings, whereas those users were probably on post 4.0. 

A simple fix would be to swap the number positions and if it === 1, then do pre 4.0. Otherwise do post 4.0.

Does this make sense? I read it on mobile so it’s possible I missed something.

zerbzman committed:
Update index.js

Changed the default firmware to be the else to make sure upgraded users would still have the default firmware.

zerbzman commented:
Gotcha, I didn't think about that. I made an update to switch the default to the 'else'. See what you think. Thanks

josephdadams commented:
Looks great! Version bump package.json to 1.1.5 and we should be good to go!

zerbzman committed:
Update package.json

Update version number

zerbzman commented:
Great!

josephdadams commented:
Tagged, released, and merged into core. Thanks!
